tests/rptest/clients/rpk.py

```diff
@@ -311,6 +311,10 @@ def parse_field(field_name, string):
 
         def parse_partition(string):
             m = partition_pattern.match(string)
+
+            if m is None and string.find('NOT_LEADER_FOR_PARTITION') != -1:
```

Comment:
Small comment about why this is needed would be good, and an example of the line format that we hit in this case.